session.go

```diff
@@ -3124,6 +3123,41 @@ func (c *Collection) Upsert(selector interface{}, update interface{}) (info *Cha
 	return info, err
 }
 
+// Same as Upsert, but sets multi to True so it can update multiple documents, but upsert if none
+// exist
+func (c *Collection) UpsertMulti(selector interface{}, update interface{}) (info *ChangeInfo, err error) {
```

Comment:
Since this is basically the same as ```Upsert```, what do you think about creating a private upsert method that takes the ```updateOp``` as parameter and handles the actual upsert, allowing to reuse that part of the code?
It will make the code more maintainable and flexible for possible future additions.